serialization/src/main/kotlin/io/kinference/protobuf/arrays/DoubleArraySerializer.kt

```diff
@@ -0,0 +1,58 @@
+package io.kinference.protobuf.arrays
+
+import io.kinference.ndarray.arrays.pointers.DoublePointer
+import io.kinference.ndarray.arrays.tiled.DoubleTiledArray
+import io.kinference.protobuf.ProtobufReader
+
+internal class DoubleArrayBuilder(private var data: DoubleArray) : PrimitiveArrayBuilder<DoubleArray>() {
+    override var position: Int = data.size
+        private set
+
+    init {
+        checkCapacity(INITIAL_CAPACITY)
+    }
+
+    override fun checkCapacity(requiredCapacity: Int) {
+        if (data.size < requiredCapacity)
+            data = data.copyOf(requiredCapacity.coerceAtLeast(data.size * 2))
```

Comment:
Can we rewind the buffer to the next field and get array size based on the size of the skipped bytes?

Reply:
No, wire protobuf reader doesn't allow that, but I tried:(